pkg/ruler/compat.go

```diff
@@ -51,11 +52,11 @@ func (a *appendableAppender) AddFast(l labels.Labels, ref uint64, t int64, v flo
 }
 
 func (a *appendableAppender) Commit() error {
-	if _, err := a.pusher.Push(a.ctx, client.ToWriteRequest(a.samples)); err != nil {
-		return err
-	}
+	ctx, cancel := context.WithTimeout(a.ctx, 15*time.Second)
```

Comment:
I think it would be better if you could mutate `a.ctx` in `Evaluate()`, then you could retain the overall timeout and other benefits of passing the Context down.